src/suites/cts/validation/createRenderPipeline.spec.ts

```diff
@@ -0,0 +1,233 @@
+  t.device.createRenderPipeline(goodDescriptor);
+
+  // Fails because 3 is not a power of 2.
+  const badDescriptor = t.getDescriptor({
+    sampleCount: 3,
+  });
+
+  await t.expectValidationError(() => {
+    t.device.createRenderPipeline(badDescriptor);
+  });
+});
+
+g.test('sample count must be equal to the one of every attachment in the render pass', async t => {
+  const { multisampledPass, multisampledPipeline, success } = t.params;
+
+  const sampleCountForTexture = multisampledPass ? 4 : 1;
+  const colorTexture = t.createTexture({
+    format: 'rgba8unorm',
+    sampleCount: sampleCountForTexture,
+  });
+  const depthStencilTexture = t.createTexture({
+    format: 'depth24plus-stencil8',
+    sampleCount: sampleCountForTexture,
+  });
+  const renderPassDescriptorWithoutDepthStencil = {
+    colorAttachments: [
+      {
+        attachment: colorTexture.createView(),
+        loadValue: { r: 1.0, g: 0.0, b: 0.0, a: 1.0 },
+      },
+    ],
+  };
+  const renderPassDescriptorWithDepthStencilOnly = {
+    colorAttachments: [],
+    depthStencilAttachment: {
+      attachment: depthStencilTexture.createView(),
+      depthLoadValue: 1.0,
+      depthStoreOp: 'store',
+      stencilLoadValue: 0,
+      stencilStoreOp: 'store',
+    },
+  };
+
+  const sampleCountForPipeline = multisampledPipeline ? 4 : 1;
+  const pipelineWithoutDepthStencilDescriptor = t.getDescriptor({
+    sampleCount: sampleCountForPipeline,
+  });
+  const pipelineWithoutDepthStencil = t.device.createRenderPipeline(
+    pipelineWithoutDepthStencilDescriptor
+  );
+  const pipelineWithDepthStencilOnlyDescriptor = t.getDescriptor({
+    colorStates: [],
+    depthStencilState: {
+      format: 'depth24plus-stencil8',
+      stencilBack: {
+        compare: 'always',
+        failOp: 'keep',
+        depthFailOp: 'keep',
+        passOp: 'keep',
+      },
+      stencilFront: {
+        compare: 'always',
+        failOp: 'keep',
+        depthFailOp: 'keep',
+        passOp: 'keep',
+      },
+    },
+    sampleCount: sampleCountForPipeline,
+  });
+  const pipelineWithDepthStencilOnly = t.device.createRenderPipeline(
+    pipelineWithDepthStencilOnlyDescriptor
+  );
+
+  for (const { renderPassDescriptor, pipeline } of [
+    {
+      renderPassDescriptor: renderPassDescriptorWithoutDepthStencil,
+      pipeline: pipelineWithoutDepthStencil,
+    },
+    {
+      renderPassDescriptor: renderPassDescriptorWithDepthStencilOnly,
+      pipeline: pipelineWithDepthStencilOnly,
+    },
+  ]) {
+    const commandEncoder = t.device.createCommandEncoder();
+    const renderPass = commandEncoder.beginRenderPass(renderPassDescriptor);
+    renderPass.setPipeline(pipeline);
+    renderPass.endPass();
+
+    await t.expectValidationError(() => {
+      commandEncoder.finish();
+    }, !success);
+  }
+}).params([
+  { multisampledPass: true, multisampledPipeline: true, success: true }, // It is allowed to use multisampled render pass and multisampled render pipeline.
```

Comment:
nit: these could just be attachmentSamples: 4, pipelineSamples: 4

Reply:
Done